codegen/model/resource.go

```diff
@@ -134,13 +134,13 @@ func (g *Group) InitDescriptors(descriptors []*collector.DescriptorWithPath) {
 // ensures the resources point to this group
 func (g *Group) Init() {
 	for i, resource := range g.Resources {
-		resource.Group = g
+		resource.Group = *g
```

Comment:
i'm not 100% sure what this might affect in case we need this somewhere else, but there is a better way to solve this bug when iterating over a slice of pointers:

```
for _, ptr := range ptrValues {
    ptr := ptr // shadow the variable ensures the correct value will be used
    // act on ptr here
}
```

Reply:
ah you're right, i had been trying to do that but didn't find the right spot. fixed now